### PR TITLE
Fixing a potential case of callback getting nullified

### PR DESCRIPTION
**Pull request by sojanpr**
Call back may get nullified if  init seen is called  before auto
session start and  if Fb  app link check is on going

@aaustin @Sarkar @EvangelosG

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/Branch.java [hunk 1]
```diff
@@ -2945,7 +2947,9 @@ private void registerInstallOrOpen(ServerRequest req, BranchReferralInitListener
         // be cleared if the app is terminated while an Open/Install is pending.
         else {
             // Update the callback to the latest one in initsession call
-            deferredInitListener_ = new WeakReference<>(callback);
+            if (callback != null) {
+                deferredInitListener_ = new WeakReference<>(callback);
+            }
             requestQueue_.setInstallOrOpenCallback(callback);
```
Comment (Sarkar) [inline Branch-SDK/src/io/branch/referral/Branch.java]: does requestQueue ignore null here?

Comment (sojanpr) [inline Branch-SDK/src/io/branch/referral/Branch.java]: Good point sahil , We already do `if (callback != null) {` check inside the method `setInstallOrOpenCallback`

**Comment by aaustin** [comment]
seems good to me 👍

**Comment by EvangelosG** [comment]
👍

**Comment by Sarkar** [comment]
👍

**Pull request merged**